promotion/datastore.go

```diff
@@ -450,28 +453,34 @@ func (pg *Postgres) GetAvailablePromotions(platform string, legacy bool) ([]Prom
 	statement := `
 		select
 			promotions.*,
+			coalesce(false, claims.legacy_claimed) as legacy_claimed,
 			true as available,
 			array_to_json(array_remove(array_agg(issuers.public_key), null)) as public_keys
 		from
-		promotions left join issuers on promotions.id = issuers.promotion_id
+		promotions 
+			left join issuers on promotions.id = issuers.promotion_id
+			left join claims on promotions.id = claims.promotion_id
 		where promotions.promotion_type = 'ugp' and
 			( promotions.platform = '' or promotions.platform = $1) and
 			promotions.active and promotions.remaining_grants > 0
-		group by promotions.id
+		group by promotions.id, claims.legacy_claimed
 		order by promotions.created_at;`
 
 	if legacy {
 		statement = `
 		select
 			promotions.*,
+			coalesce(false, claims.legacy_claimed) as legacy_claimed,
```

Comment:
since this is purely a mechanism to alert clients using the new promotion endpoint that they can re-claim a legacy claimed promotion, we can just always return `false` for `legacy_claimed` here